### ISSUE TITLE
Question regarding releasing git2.dll resources, and then reloading them.

### ISSUE THREAD

**Issue by XaeroDegreaz** [libgit2/libgit2sharp#341]
I'm using this library in a Unity project. For those unfamiliar with Unity, when it loads a plugin (git2.dll, in this case) it stays loaded in a write-locked stated until you restart the editor, or make changes to your assembly.

Unity has a nifty feature which basically reloads the assembly when it detects modifications from the outside, however I get random crashes when it tries to do a reload of the git2.dll. It's tricky to pin-point where it's coming from, but I'm sure it has to do with the dll file being locked in some way, and Unity not being able to release it fast enough, causing an `Access Violation`.

In any case, I've stumbled across `LoadLibrary()` and `FreeLibrary()` methods found in `kernel32.dll` which seem to be a step in the right direction. However, `FreeLibrary()` takes an `IntPtr` parameter that references the aforementioned loaded library, and I see nowhere in the `NativeMethods` class where there is any such reference made (presumably because this is a cross-platform lib, and `kernell32.dll` is Windows only).

Basically, I'm trying to figure out the best way to load the plugin manually, so that I can keep a reference, and then unload the plugin after select `libgit2` calls, so Unity doesn't have a reason to crash when making changes to the assembly after the plugin has been loaded.

Is this even possible without completely hacking deeply into the LibGit2Sharp code, and custom writing methods to call functions on manually loaded library references? Is this even the right word?

I apologize for my ignorance, but I've never actually dealt so closely with programming at this low of a level. If there's any other information that I could provide to explain my intentions further, please let me know.
